src/pow/miner.cpp

```diff
@@ -102,22 +102,19 @@ void Miner::Run() {
             Block b(GetParams().version);
 
             if (!selfChainHead_) {
-                auto firstRegTx = MEMPOOL->GetRedemptionTx(true);
-                if (!firstRegTx) {
-                    spdlog::info("[Miner] Paused. Waiting for the first registration.");
-                    while (enabled_ && !firstRegTx) {
-                        std::this_thread::yield();
-                        firstRegTx = MEMPOOL->GetRedemptionTx(true);
-                    }
-                }
+                spdlog::info("[Miner] Paused. Waiting for the first registration...");
+                ConstTxPtr firstRegTx;
+                while (enabled_ && !(firstRegTx = MEMPOOL->GetFirstReg()))
```

Comment:
can we add a sleep mechanism?

Reply:
It’s not necessary here since we use BlockingQueue